app/controllers/application_controller.rb

```diff
@@ -9,4 +9,12 @@ def configure_permitted_parameters
     devise_parameter_sanitizer.permit(:sign_up, keys: attributes)
     devise_parameter_sanitizer.permit(:account_update, keys: attributes)
   end
+
+  def authenticate_to_submit
+    if !user_signed_in?
+      flash[:alert] = "You must be logged in to submit a post or comment"
+      redirect_to new_user_session_path
+    end
+  end
```

Comment:
Might be worth tucking this in a `Submitable` module shared between the Posts and Comments controllers.

Reply:
Hmm - making this issue here: https://github.com/gingin77/hackernews_clone/issues/17